### ISSUE TITLE
Poison recipe

### ISSUE THREAD

**Issue by vandervecken** [kevL/TccScripts#7]
This recipe is labeled as a general on-hit property (ID 48) in the text file, but it’s actually a on-hit cast ‘Poison’ spell property (ID 82) with CL 4 (Large Scorpion Venom: 1d4 Con damage on primary and secondary hit, DC 14).

    Poisoned Weapon: Poison (DC 14, 1d2 Con Damage)
    Requires: Caster Level 10, Craft Magic Arms and Armor feat
    Ingredients: Weak Earth Essence + Emerald
    Spell: Poison

**Comment by Aqvilinus** [comment]
What I meant here is that there're two groups of on-hit item properties: producing general effects (like wounding, silence, doom, slow, etc; see the X1006 section) and casting spell (see the X1002 section). The recipe I mentioned above is defined as the cast 'Poison' spell (CL 4) property in crafting.2da. What I find confusing here is that on-hit: cast 'Poison' spell is also among available armor spells in the X1002 section, but not among availabel weapon spells. So you can apply, for example, on-hit cast 'Poison' spell (CL 20) to your armor, but only on-hit cast 'Poison' spell (CL 4) to your weapon. Is this intended?

**Comment by Aqvilinus** [comment]
What I meant here is that there're two groups of on-hit item properties: producing general effects (like wounding, silence, doom, slow, etc; see the X1006 section) and casting spell (see the X1002 section). The recipe I mentioned above is defined as cast 'Poison' spell (CL 4) property in crafting.2da. What I find confusing here is that 'Poison' spell is also among available armor spells in the X1002 section, but not among weapon spells. So you can apply, for example, on-hit cast 'Poison' spell (CL 20) to your armor, but only on-hit cast 'Poison' spell (CL 4) to your weapon. Is this intended? I guess this was done so because the general on-hit: poison property was bugged and didn't work (?) and it was replaced by this spell version for weapons.

**Comment by vandervecken** [comment]
Github had some issues so not sure if notifications went out. @kevL and @Aqvilinus ^^

**Comment by kevL** [comment]
1st time i ever seen github go wonky ... must've had something to do with the M$ buyout hehe

**Comment by vandervecken** [comment]
Has anyone managed to test this?

**Comment by vandervecken** [comment]
@kevL @Aqvilinus @zzcmeiou can anyone test this?

**Comment by kevL** [comment]
![yata_itemprodef](https://user-images.githubusercontent.com/5395276/47873316-8346df80-ddd6-11e8-8d21-b3d5ec317be2.png)

**Comment by kevL** [comment]
I tried this on (a) a dagger, (b) a robe

```c++
ip = ItemPropertyOnHitProps(IP_CONST_ONHIT_ITEMPOISON, IP_CONST_ONHIT_SAVEDC_20, IP_CONST_POISON_1D2_STRDAMAGE);
AddItemProperty(DURATION_TYPE_PERMANENT, ip, oTarget, 0.f);
```

it stuck on the dagger, appearing on the Examine sheet
it did NOT stick to the robe.

- looks like that's the reason OnHitCastSpell(poison) is applied on armors ...
- it worked against a goblin ... lowered its strength
- didn't check any other OnHitProps (they might not work on armors?)

But this worked on both dagger and robe:

```c++
ip = ItemPropertyOnHitCastSpell(IP_CONST_ONHIT_CASTSPELL_POISON, 20);
AddItemProperty(DURATION_TYPE_PERMANENT, ip, oTarget, 0.f);
```
/tested IG

**Comment by vandervecken** [comment]
I'm finding myself still confused by this @kevL. Instead of asking you to explain again what is happening here, I'll just ask:
do you have a suggested remediation?

**Comment by kevL** [comment]
```
recipe ids #265, 266, 267 are (ip 82) OnHitCastSpell:Poison @ L12,16,20 (for armor/shield)
recipe ids #262, 263, 264 are (ip 82) OnHitCastSpell:Poison @ L 4, 4, 8 (#262 for melee, #263,264 for armor/shield)
```

My test indicates they all should work. The issue, i believe, is in how they are listed in the Recipe Manual (that the player reads and uses to find recipes). I don't know where/how they are currently, but they should appear in the OnHitCastSpell section only. That is, not in "CastSpell" nor in "OnHit"

they are "OnHitCastSpell"

(note: I don't see any OnHit:ItemPoison recipes in the 2da...)

ps. I think Aqvilinus' original issue is that there's no recipe for L8 on melee weapons (nor for L12,16,20). That is, there appears to be some sort of imbalance between poison recipes for armor/shield and melee weapons ...

sounds like a policy decision ;)

**Comment by kevL** [comment]
I think this is a documentation issue ...

**Comment by kevL** [comment]
there's two ways of applying a Poison ip

1. ItemPropertyOnHitProps()
2. ItemPropertyOnHitCastSpell()

either can be applied to (a) a weapon, (b) an armor

1. works on weapons but does NOT work on armors (bug, at least for Poison)
2. works on both weapons and armor

I believe that all the recipes in the 2da are valid. I can't speak to the documentation but you should probly check that poison recipes are described under their proper categories in the Readme.

@Aqvilinus fundamental issue appears to be this:
>So you can apply, for example, on-hit cast 'Poison' spell (CL 20) to your armor, but only on-hit cast 'Poison' spell (CL 4) to your weapon.

ie, that the power of the OnHitCastSpell(poison) goes up to L20 on armor, but only to L4 for a weapon.

**Comment by vandervecken** [comment]
Ok, so can we confirm the following?
* The OnHitProps recipe when applied to a Dagger (DC 14 etc) works
* There is No OnHitProps recipe that exists in the 2da for armor, and none such can be made
* The OnHitCastSpell recipe goes up to CL20 for Armor and works correctly.
* There are missing recipes for OnHitCastSpell for weapons, and I can just go and make those, they will work fine.

Is that right?

**Comment by kevL** [comment]
sounds (exactly) right.

**Comment by vandervecken** [comment]
No, it's not quite right.
The current recipe referenced in the very first post in this thread is *supposed* to generate an OnHitProps version of the effect, but it does not. That needs to change.

**Comment by vandervecken** [comment]
Got it, 48.19.0.2.0 - thanks Google.

**Comment by kevL** [comment]
heh, for future reference:

OnHitProp,Poison,DC14,d2Con

48,19,0,2,0

48 - onhitprop ( ItemPropDef.2da )
19 - poison ( Iprp_OnHit.2da )
0 - dc14 ( nwscript constant )
2 - d2 con damage ( nwscript constant )
0 - not used.


itemproperty ItemPropertyOnHitProps(int nProperty, int nSaveDC, int nSpecial=0); 48, ITEM_PROPERTY_ON_HIT_PROPERTIES
nProperty = 19, IP_CONST_ONHIT_ITEMPOISON
nSaveDC = 0, IP_CONST_ONHIT_SAVEDC_14
nSpecial = 2, IP_CONST_POISON_1D2_CONDAMAGE


y, it's a b*tch to figure out... that's why I did Yata. Literally, this is why I coded Yata, with extended Info for Crafting.2da

in fact i worked backwards to get the above information; first I got the info for the crafting IP to display on Yata's statusbar, then i figured out why those values appeared (i actually had to look at the sourcecode for Yata to figure it out...)

The best way, however, to figure out an encodedIp is to look at 'x2_inc_itemprop' IPGetItemPropertyByID() -- that function lists all available IP-constructors (is used by crafting). Once you have the name of the constructor, look it up in NwScript.nss, where there's usually enough information in the function's description to figure things out
